c

```diff
@@ -0,0 +1,259 @@
+#define _CRT_SECURE_NO_WARNINGS
+#include <iostream>
+#include <stdio.h>
+#include <string.h>
+
+typedef struct radnici {
+	int sifra;
+	char imeRadnika[21];
+	char prezimeRadnika[21];
+	float koeficijenat;
+	int radniStaz;
+}tipRadnika;
+
+
+void unosRadnika(tipRadnika* pradnik) {
+	puts("Unesi podatke o radniku:");
+	printf("Unesi sifru radnika:\n");
+	scanf("%d", &pradnik->sifra);
+	puts("Unesi ime radnika:");
+	scanf("%s", &pradnik->imeRadnika);
+	puts("Unesi prezime radnika:");
+	scanf("%s", &pradnik->prezimeRadnika);
+	puts("Unesi koeficijenat radnika:");
+	scanf("%f", &pradnik->koeficijenat);
+	printf("Unesi radni staz:\n");
+	scanf("%d", &pradnik->radniStaz);
+}
+
+void prikaziRadnika(tipRadnika radnik)
+{
+	puts("\nPodaci o radniku:");
+	printf("Sifra radnika: %d\n", radnik.sifra);
+	printf("Ime radnika: %s\n", radnik.imeRadnika);
+	printf("Prezime radnika: %s\n", radnik.prezimeRadnika);
+	printf("Koeficijenat radnika: %.2f\n", radnik.koeficijenat);
+	printf("Radni staz radnika: %d\n", radnik.radniStaz);
+}
+
+void brisiRadnika(int* n, tipRadnika radnici[], int redniBr) {
+	int i, j = 0;
+	for (i = redniBr - 1; i < *n - 1; i++)
+	{
+		radnici[i] = radnici[i + 1];
+	}
+	(*n)--;
+}
+
+void upisiRadnike(tipRadnika radnik[], int brRadnika) {
+	FILE* fpr = fopen("radnici.txt", "w");
+	if (fpr == NULL) {
+		printf("Greska pri otvaranju fajla\n");
+		return;
+	}
+	for (int i = 0; i < brRadnika; i++) {
+		fprintf(fpr, "%d %s %s %.2f %d\n", radnik[i].sifra, radnik[i].imeRadnika, radnik[i].prezimeRadnika, radnik[i].koeficijenat, radnik[i].radniStaz);
+	}
+	fclose(fpr);
+}
+
+void procitajRadnike(FILE* pfradnik, tipRadnika* radnik)
+{
+	fscanf(pfradnik, "%d", &radnik->sifra);
+	fscanf(pfradnik, "%s", radnik->imeRadnika);
+	fscanf(pfradnik, "%s", radnik->prezimeRadnika);
+	fscanf(pfradnik, "%.2f", &radnik->koeficijenat);
+	fscanf(pfradnik, "%d", &radnik->radniStaz);
+}
+
+void kopiraj_u_fajl(FILE* pftxt, tipRadnika radnik)
+{
+	fprintf(pftxt, "%d\n", radnik.sifra);
+	fprintf(pftxt, "%s\n", radnik.imeRadnika);
+	fprintf(pftxt, "%s\n", radnik.prezimeRadnika);
+	fprintf(pftxt, "%f\n", radnik.koeficijenat);
+	fprintf(pftxt, "%d\n", radnik.radniStaz);
+}
+
+void izmeniRadnika(tipRadnika* radnik, int odredjenaSifra) {
+
+	printf("Izmena podataka za radnika sa sifrom %d\n", radnik->sifra);
+	printf("Unesi novo ime radnika: \n");
+	scanf("%s", radnik->imeRadnika);
+	printf("Unesi novo prezime radnika: \n");
+	scanf("%s", radnik->prezimeRadnika);
+	printf("Unesi novi koeficijenat radnika: \n");
+	scanf("%f", &radnik->koeficijenat);
+	printf("Unesi novi radni staz radnika: \n");
+	scanf("%d", &radnik->radniStaz);
+
+}
+
+bool trazeniKoeficijenat(tipRadnika radnik, float trazeniKoef) {
+	return radnik.koeficijenat == trazeniKoef;
+}
+
+float izracunajTrenutnuZaradu(tipRadnika radnik, float trenutnaCenaRada) {
+	return radnik.koeficijenat * trenutnaCenaRada * (1 + radnik.radniStaz * 0.5 / 100);
```

Comment:
Obzirom da imas deo gde kazes: `x * 0.5 / 100` a mnozenje sa 0.5 je kao deljenje sa 2, to mozes da uprostis na sledece, posto te matematicke operacije imaju istu prednost:

```suggestion
	return radnik.koeficijenat * trenutnaCenaRada * (1 + radnik.radniStaz / 200);
```

Reply:
Nije mi palo na pamet : o